File: Day1/first.py
## first.py

def process_line(line):
    numbers = [int(s) for s in line if s.isdigit()]  # Extract numbers from the line
    print(f"Numbers: {numbers}")
    if numbers:  # Check if numbers are present
        first_number = numbers[0]
        last_number = numbers[-1]
        calibration_value = str(first_number) + str(last_number)
        print(f"Clibration Value: {calibration_value}")
        print(first_number + last_number)
        return int(calibration_value)
    return 0  # Return 0 if no numbers found in the line


def read_file(file_path):
    try:
        with open(file_path, 'r') as file:
            calibration_list = []
            counter = 0
            for line in file:
                counter += 1
                calibration_value = process_line(line)
                if calibration_value != 0:
                    calibration_list.append(calibration_value)
                
            print(f"sum, {sum(calibration_list)}")
            return sum(calibration_list)
            
    except FileNotFoundError:
        print("File not found.")
    except IOError as e:
        print("Error reading the file:", e)

# Replace 'path/to/your/file.txt' with the actual file path
file_path = './inputD1n.txt'
read_file(file_path)

File: Day1/first.js
"use strict";
// The newly-improved calibration document consists of lines of text; each line originally contained a specific calibration value that the Elves now need to recover. On each line, the calibration value can be found by combining the first digit and the last digit (in that order) to form a single two-digit number.
Object.defineProperty(exports, "__esModule", { value: true });
//For example:
// 1abc2
// pqr3stu8vwx
// a1b2c3d4e5f
// treb7uchet
// In this example, the calibration values of these four lines are 12, 38, 15, and 77. Adding these together produces 142.
//
// Consider your entire calibration document. What is the sum of all of the calibration values?
//
var fs = require("fs");
var filePath = "path/to/your/file.txt"; // Replace with your file path
fs.readFile(filePath, "utf8", function (err, data) {
    if (err) {
        console.error("Error reading file:", err);
        return;
    }
    console.log("File content:", data);
});
